### ISSUE TITLE
Prototype Question Repository Page

### ISSUE THREAD

**Issue by CaptainAchilles** [hkhosrav/RiPLE-Client-side#3]
Part of this project is to allow users to see the full question repository on a page and be able to search them. A prototype should be implemented to get a feel for this.

Some data that exists for questions is:
```
    id: number,
    rating: number,
    responseCount: number,
    difficulty: number,

    topic: string,
    title: string,
    content: string
```

**Comment by CaptainAchilles** [comment]
This has been implemented on improvement-#2 with questions populated with random data

**Comment by hkhosrav** [comment]
I like the idea of using cards (grids) for presenting the questions. Would it be computationally  expensive to show a preview of the questions on the cards?

**Comment by CaptainAchilles** [comment]
`question-prototype-#3` has been opened to continue this ticket.

I don't think so since the question preview is quite easy to calculate. We can just render as much content which fits in the display area of the card preview.

What sorts of information is important to show the user as a preview? We have the following data which we could show:

```
id: number,
topic: string,
content: string,
authorsReputation: string,
numberOfAnswers: number,
authorsAnswersMostPopular: boolean,
helpRequests: number // What is this?
mostRecentComments: number // What is this?
numberOfComments: number,
difficulty: number,
quality: number
```

I think that at a glance students would be interested in seeing:
* Preview
* Question Quality
* Question Difficulty
* Number Answers

**Comment by CaptainAchilles** [comment]
@hkhosrav I am not too sure on the following fields for peerWise, would you be able to explain them?

* Help Requests
* Most Recent Comments (they all seem to be set to '-' on the peerwise page)

**Comment by hkhosrav** [comment]
I think we can ignore the fields associated with Help Requests and Most Recent Comments

**Comment by CaptainAchilles** [comment]
As per out in-person discussions, the cards have been redesigned a bit. Most notably:
* The Visulisations have been moved to the bottom of the card
* Images are now supported
* Increased card height
* Added multiple topics

![image](https://user-images.githubusercontent.com/16090868/27058519-2375fb18-5015-11e7-9a45-d40840afa334.png)
![image](https://user-images.githubusercontent.com/16090868/27058529-3592fd64-5015-11e7-8d53-e1457ade0f6f.png)

**Comment by CaptainAchilles** [comment]
Alternative design mockup:
![image](https://user-images.githubusercontent.com/16090868/27059178-801504b0-5018-11e7-8dac-6e64071474d5.png)

**Comment by CaptainAchilles** [comment]
Another alternative design mockup
![image](https://user-images.githubusercontent.com/16090868/27059264-ec0b1236-5018-11e7-919c-a6039d78f93e.png)

**Comment by CaptainAchilles** [comment]
Slight variation on previous:
![image](https://user-images.githubusercontent.com/16090868/27059287-1661596e-5019-11e7-8638-8dbe1d8db571.png)

**Comment by hkhosrav** [comment]
Looking good! Could we try the rating visualisations with http://vuejsexamples.com/a-collection-of-circle-progress-with-vue-js/ ?

Also the Number Answers logo is often used for showing the number of comments. Let's see if we can find a replacement.

**Comment by CaptainAchilles** [comment]
![image](https://user-images.githubusercontent.com/16090868/27068836-57f64c0e-5055-11e7-85fb-95423f6f13df.png)

I moved all the card info into the bottom. 
* Changed the `comments` logo
* Balanced the information

**Comment by CaptainAchilles** [comment]
I am thinking instead of cutting the text off with an `...` we could apply a fade towards the truncation point of the text.

**Comment by hkhosrav** [comment]
Great work! looks really good. I think let's close the issue and move on for the time being. We'll revisit the repository page once we have a better sense of the other pages.

**Comment by solmazabdi** [comment]
It looks really good Nich!!!

**Comment by CaptainAchilles** [comment]
Closed 253afd0ef5bd674e972526cbeeec749df5410653
